TS/src/BenchmarkRunner.lf

```diff
@@ -0,0 +1,117 @@
+target TypeScript;
+
+/*
+ *
+ * Copyright (C) 2022 TU Dresden and UC Berkeley
+ *
+ * @author Wonseo Choi
+ */
+
+reactor BenchmarkRunner(num_iterations:number(12)) {
+
+    output start:boolean;
+    input finish:boolean;
+    
+    logical action nextIteration:boolean;
+    logical action done:boolean;
+    
+    state count:number(0);
+    
+    /** Start time for runtime measurement. */
+    state startTime:time;
+    
+    /** Runtime measurements. */
+    //TODO : "any" Array ! -> change another function code //
+    state measuredTimes: {=Array<any>=}({= [] =});
+    
+    
+    reaction(startup) -> nextIteration {=
+        // Initialize an array of interval_t
+        measuredTimes.length = num_iterations;
+        actions.nextIteration.schedule(0, true);
+        console.log("Benchmark will run " + num_iterations + " times \n");
+    =}
+   
+    
+    reaction(nextIteration) -> start, done {=
+        if (count < num_iterations) { 
+            startTime = util.getCurrentPhysicalTime();
+            start = true;
+        } else {
+            actions.done.schedule(0, true);
+        }
+    =}
+    
+    reaction(finish) -> nextIteration {=
+        const duration = util.getCurrentPhysicalTime().subtract(startTime as TimeValue);
+        measuredTimes[count] = duration.toMilliseconds();
+        ++count;
+        
+        console.log("Iteration "+ count + " - " + duration.toMilliseconds() + " ms\n");
+        
+        actions.nextIteration.schedule(0, true);
+        
+    =}
+    
+    reaction(done) {=        
+        const measuredMSTimes:number[] = getMSMeasurements(measuredTimes, num_iterations);
+        measuredMSTimes.sort()
+        
+        console.log("Execution - Summary:\n");
+        console.log("Best Time:\t " + measuredMSTimes[0] + " msec\n");
+        console.log("Worst Time:\t " + measuredMSTimes[num_iterations - 1] + " msec\n");
+        console.log("Median Time:\t " + median(measuredMSTimes, num_iterations) + " msec\n");
+        
+        util.requestStop();
+    =}
+    
+    preamble {=
+
+        function median(execTimes: Array<number>, size: number): number {
+            if (size == 0) {
+                return 0.0;
+            }
+
+            const middle = size / 2;
+
+            if(size % 2 == 1){
+                return execTimes[middle]
+            } else {
+                return (execTimes[middle-1] + execTimes[middle]) / 2;
+            }
+
+        }
+
+        function getMSMeasurements(measured_times: Array<any>, num_iterations: number): number[] {
+            const msMeasurements = new Array<number>(num_iterations);
+            for (let i = 0; i < num_iterations; i++) {
+                msMeasurements[i] = measured_times[i];
+            }
+
+            return msMeasurements;
+        }
+        
+    =}
+    
+    preamble {=
+        
+        // TODO : Implement OS infrom !! // 
```

Comment:
There's no `window` object in node, so I'm unclear as to what this is meant to do...